perllib/FixMyStreet/DB/Result/Problem.pm

```diff
@@ -261,6 +261,21 @@ with 'FixMyStreet::Roles::Abuser',
      'FixMyStreet::Roles::Translatable',
      'FixMyStreet::Roles::PhotoSet';
 
+before update => sub {
```

Comment:
What are the situations where a problem's location may be updated? I think there are two (may have forgotten others!) - when a report is edited in the admin, and when its pin is dragged in the inspector form. I think rather than having this method on the `Problem` model it would be better to use/create a small cobrand hook that can make changes to a problem before it's saved. I think that would allow you to remove the `land_type_for_problem` method in `Default.pm` too.